client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -169,16 +169,23 @@ object RetryPolicy {
     GatewayTimeout,
   )
 
-  /** Returns true if (the request method is idempotent or request contains Idempotency-Key header)
-    * and the result is either a throwable or has one of the `RetriableStatuses`.
+  /** Returns true if (the request method is idempotent and the request's entity is pure (isn't effectful);
+    * or request contains [[`Idempotency-Key`]] header) and the result is either a throwable or
+    * has one of the `RetriableStatuses`.
     *
-    * Caution: if the request body is effectful, the effects will be
-    * run twice.  The most common symptom of this will be resubmitting
-    * an idempotent request.
+    * Caution: a request wouldn't be resubmitted if the idempotent request contains
+    * streamed (effectful) [[Entity]] (nor empty, nor fully loaded into memory).
```

Comment:
I'm not sure if we need this "caution" anymore, I think these changes make this method safer.

Reply:
Yep, but this changes behavior without changes in API. So maybe it's worth leaving it, I dunno..